addons/html_editor/static/tests/power_buttons.test.js

```diff
@@ -0,0 +1,123 @@
+import { describe, expect, test } from "@odoo/hoot";
+import { click, press, waitFor } from "@odoo/hoot-dom";
+import { animationFrame } from "@odoo/hoot-mock";
+import { setupEditor } from "./_helpers/editor";
+import { getContent } from "./_helpers/selection";
+import { insertText } from "./_helpers/user_actions";
+import { onRpc } from "@web/../tests/web_test_helpers";
+
+describe.tags("desktop")("visibility", () => {
+    test("should show power buttons on empty P tag", async () => {
+        const { editor } = await setupEditor("<p>[]<br></p>");
+        expect(".o_we_power_buttons").toBeVisible();
+        insertText(editor, "a");
+        await animationFrame();
+        expect(".o_we_power_buttons").not.toBeVisible();
+    });
+
+    test("should not show power buttons on heading tags", async () => {
+        await setupEditor("<h1>[]<br></h1>");
+        expect(".o_we_power_buttons").not.toBeVisible();
+    });
+
+    test("should not show power buttons in table", async () => {
+        await setupEditor(
+            "<table><tbody><tr><td><p>[]<br></p></td><td><p><br></p></td></tr></tbody></table>"
+        );
+        expect(".o_we_power_buttons").not.toBeVisible();
+    });
+
+    test("should not show power buttons in banner", async () => {
+        const { editor } = await setupEditor("<p>[]<br></p>");
+        await waitFor(".o_we_power_buttons");
+        insertText(editor, "/banner");
+        press("enter");
+        await animationFrame();
+        expect(".o_we_power_buttons").not.toBeVisible();
+    });
+
+    test("should not show power buttons in columns", async () => {
+        const { editor } = await setupEditor("<p>[]<br></p>");
+        expect(".o_we_power_buttons").toBeVisible();
+        insertText(editor, "/columns");
+        press("enter");
+        await animationFrame();
+        expect(".o_we_power_buttons").not.toBeVisible();
+    });
+
+    test("should not show power buttons on empty p tag with text-align style", async () => {
+        await setupEditor('<p style="text-align: right;">[]<br></p>');
+        expect(".o_we_power_buttons").not.toBeVisible();
+    });
+
+    test("should not show power buttons on non empty block P tag", async () => {
+        await setupEditor("<p>[]<br><br></p>");
+        expect(".o_we_power_buttons").not.toBeVisible();
+    });
+});
+
+describe.tags("desktop")("buttons", () => {
+    test("should create a numbered list using power buttons", async () => {
+        const { el } = await setupEditor("<p>[]<br></p>");
+        click(".o_we_power_buttons .power_button.fa-list-ol");
```

Comment:
You have rendered all power_button action asynchronous, and this test does not await anything